openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -1207,6 +1207,7 @@ This specification defines the following Credential Issuer Metadata:
 * `credential_response_encryption_enc_values_supported`: OPTIONAL. Array containing a list of the JWE [@!RFC7516] encryption algorithms (`enc` values) [@!RFC7518] supported by the Credential and/or Batch Credential Endpoint to encode the Credential or Batch Credential Response in a JWT [@!RFC7519].
 * `require_credential_response_encryption`: OPTIONAL. Boolean value specifying whether the Credential Issuer requires additional encryption on top of TLS for the Credential Response and expects encryption parameters to be present in the Credential Request and/or Batch Credential Request, with `true` indicating support. When the value is `true`, `credential_response_encryption_alg_values_supported` parameter MUST also be provided.  If omitted, the default value is `false`.
 * `credential_identifiers_supported`: OPTIONAL. Boolean value specifying whether the Credential Issuer supports returning `credential_identifiers` parameter in the `authorization_details` Token Response parameter, with `true` indicating support. If omitted, the default value is `false`.
+* `signed_credential_issuer_metadata`: OPTIONAL. String that is an entire signed JWT. This JWT can contain Credential Issuer metadata parameters as claims. The signed metadata MUST be digitally signed or MACed JWS using JSON Web Signature (JWS) [@!RFC7515] and MUST contain an `iat` (Issued At) claim and an `iss` (Issuer) claim denoting the party attesting to the claims in the signed metadata. This JWT can also be used to identify the actual Credential Issuer when it is using third party service provider as a domain to host its metadata file, in which case, the JWT SHOULD contain `sub` (Subject) claim denoting the third party service provider. If the Wallet supports signed metadata, metadata values conveyed in the signed JWT MUST take precedence over the corresponding values conveyed using plain JSON elements. A `signed_credential_issuer_metadata` metadata value SHOULD NOT appear as a claim in the JWT. Because metadata is only partially signed, the usage of DPoP [@!RFC9449] of MTLS [@!RFC8705] is RECOMMENDED to prevent an attacker from modifying the value of the Credential Endpoint to steal Access Tokens.
```

Comment:
I didn't get this 

> Because metadata is only partially signed, the usage of DPoP [@!RFC9449] of MTLS [@!RFC8705] is RECOMMENDED to prevent an attacker from modifying the value of the Credential Endpoint to steal Access Tokens.

can we add more clarification for this?

Since there may be some cases where the metadata attester and signer is different from the credential issuer, in the vest of the Trust Third Party like it could be a National Accreditation Body, in this case we must explode a sublist with:

- signed_credential_issuer_metadata_uri can be possibile, then providing the metadata by reference
- when metadata iss is a trusted third party, the sub MUST be included. The exp should be present as well, because a signed artifact should have an expiration time.

all good, even if I have to admit that's not clear how to establish the trust to the metadata issuer ... yes, out of scope here (but still in scope in our minds).

Reply:
what I meant was tha tif the signed_metadata JWT does not contain `authorization_servers` and `credential_issuer` claims, and they are in the JSON object, an attacker can insert attacker's `credential_issuer` value and trick the wallet to send an access token from a good AS to attacker's RS, so that attacker can use it at the good RS. ping @danielfett if he could suggest a better language..?

> The exp should be present as well, because a signed artifact should have an expiration time.
I am not sure what `exp` time I would put as an issuer to the metadata. what would cause the the issuer metadata to expire..?

> that's not clear how to establish the trust to the metadata issuer ...
I agree that's something we will need to reiterate on later. for now, the goal is to put a general purpose mechanism that implementers can start using.

Reply:
I don't think we should be putting this quite general recommendation around usage of DPoP in the definition of a credential issuer metadata parameter. I think we should shift this to a seperate security consideration.

Reply:
_Memento OpenID Federation_

An `openid_credential_issuer` metadata derived from a trust chain actually doesn't bring all these worries about how to establish the trust with the metadata "issuer" or attester :-)

Reply:
I agree with Tobias that DPoP/MTLS should be general consideration that we can point to here. The focus in the description should be to say that DPoP/MTLS prevent the misuse of tokens after they are sent to manipulated (attacker-controlled) endpoints.

Another note: In the sentence 'A `signed_credential_issuer_metadata` metadata value SHOULD NOT appear as a claim' I would make that a MUST NOT as otherwise the processing is undefined.

Reply:
> Another note: In the sentence 'A signed_credential_issuer_metadata metadata value SHOULD NOT appear as a claim' I would make that a MUST NOT as otherwise the processing is undefined.

@danielfett why so..?